kontrol-service/engine/docker.go

```diff
@@ -259,21 +327,60 @@ func newStatefulPluginsAndExternalServicesFromServiceConfig(serviceConfig apityp
 	return serviceStatefulPlugins, externalServices, externalServiceDependencies, nil
 }
 
-func newClusterTopologyServiceFromServiceConfig(serviceConfig apitypes.ServiceConfig, version string) resolved.Service {
+func newClusterTopologyServiceFromServiceConfig(
+	serviceConfig apitypes.ServiceConfig,
+	deploymentConfig *apitypes.DeploymentConfig,
+	statefulSetConfig *apitypes.StatefulSetConfig,
+	version string,
+) (resolved.Service, error) {
 	service := serviceConfig.Service
-	deployment := serviceConfig.Deployment
+	serviceName := service.GetObjectMeta().GetName()
+
+	if deploymentConfig == nil && statefulSetConfig == nil {
+		logrus.Warnf("Service %s has no workload", serviceName)
+	}
+
+	if deploymentConfig != nil && statefulSetConfig != nil {
+		workloads := []string{
+			deploymentConfig.Deployment.GetObjectMeta().GetName(),
+			statefulSetConfig.StatefulSet.GetObjectMeta().GetName(),
+		}
+		logrus.Error("Service %s is associated with more than one workload: %v", serviceName, workloads)
+	}
+
 	serviceAnnotations := service.GetObjectMeta().GetAnnotations()
 
 	clusterTopologyService := resolved.Service{
-		ServiceID:      service.GetObjectMeta().GetName(),
-		Version:        version,
-		ServiceSpec:    &service.Spec,
-		DeploymentSpec: &deployment.Spec,
+		ServiceID:   service.GetObjectMeta().GetName(),
+		Version:     version,
+		ServiceSpec: &service.Spec,
+	}
+
+	if deploymentConfig != nil {
+		workload := kardinal.NewDeploymentWorkloadSpec(deploymentConfig.Deployment.Spec)
+		clusterTopologyService.WorkloadSpec = &workload
+	}
+	if statefulSetConfig != nil {
+		workload := kardinal.NewStatefulSetWorkloadSpec(statefulSetConfig.StatefulSet.Spec)
+		clusterTopologyService.WorkloadSpec = &workload
+	}
+
+	if clusterTopologyService.WorkloadSpec == nil {
+		return clusterTopologyService, stacktrace.NewError("Service %s has no workload", serviceName)
 	}
+
+	// Set default for IsStateful to true if the workload is a StatefulSet, otherwise false
+	clusterTopologyService.IsExternal = clusterTopologyService.WorkloadSpec.IsStatefulSet()
```

Comment:
`IsExternal` is ok here? or did you want to set `IsStateful` here?

Reply:
Good catch!!! It's `IsStateful`!!!